### PR TITLE
Add vscode configuration support

### PR DESCRIPTION
**Pull request by mikee47**
See documentation https://smingdev.readthedocs.io/en/feature-vscode-config/tools/vscode.html

See #1784

### CODE REVIEW

**Comment by mikee47** [comment]
> 
> 
> This pull request **introduces 1 alert** when merging [3b81792](https://github.com/SmingHub/Sming/commit/3b817923c4da035fa30d8988a0dcedd9e05d24d9) into [3d3ddab](https://github.com/SmingHub/Sming/commit/3d3ddabb6f8920e41c5497d212ef954ab99c47b9) - [view on LGTM.com](https://lgtm.com/projects/g/SmingHub/Sming/rev/pr-1e0fc612464330c69abe3e8093ec57d70b019bdf)
> 
> **new alerts:**
> 
>     * 1 for Wrong name for an argument in a call

This relates to the `exist_ok` parameter in `os.makedirs()` call. This parameter was added in python3.x so looks like LGTM is using signatures from python2; perhaps this should be changed?

File: Sming/project.mk [hunk 1]
```diff
@@ -430,6 +430,23 @@ $(foreach c,$(BUILDABLE_COMPONENTS),$(eval $(call GenerateComponentTargets,$c)))
 # The arch-specific targets, including link and firmware image creation
 include $(ARCH_BASE)/app.mk
 
+.PHONY: vscode
```
Comment (slaff) [inline Sming/project.mk:433]: In the future we could have support for more Integrated Development Environments (IDE)s. How about:
1. Adding to `make help` a new topic called "IDE tools"
2. Rename the `vscode*` targets to `ide-vscode`. This way the autocompletion in bash can hint the user to find easily all supported `ide` operations by typing `make ide[PRESS-TAB]`

**Comment by mikee47** [comment]
@slaff Are you able to try out this on a proper linux box? Specifically, the integrated debugging.

**Comment by slaff** [comment]
> Are you able to try out this on a proper linux box? Specifically, the integrated debugging.

I can try the Host debugging and will let you know.

**Comment by mikee47** [comment]
@slaff Thanks. esp8266 debugging has been discussed in past issues but no idea whether the same setup actually works on Linux or not. As noted in documentation, there are also issues with the debug output not being 'clean' but these things can be addressed as required I guess.

**Comment by slaff** [comment]
Host debugging works very well. 

image

**Comment by slaff** [comment]
Unfortunately I did not have success with the Esp8266 debugger. Getting back: `Remote replied unexpectedly to 'vMustReplyEmpty'`.  Might be related to the newer version of gcc and SDK. This could be the answer: https://github.com/pulp-platform/debug_bridge/issues/1#issuecomment-258374263

**Comment by mikee47** [comment]
I use this with WSL and works with current tools

https://github.com/SmingHub/Sming/blob/61dc191d16e5148b016fd3672350ecbb680b4061/Tools/WSL/tcp_serial_redirect.py

**Pull request merged**